src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -274,6 +275,21 @@ public static T[] AsArray<T>(this IEnumerable<T> source)
             return builder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Maps an immutable array through a function that returns ValueTask, returning the new ImmutableArray.
+        /// </summary>
+        public static async ValueTask<ImmutableArray<TResult>> SelectAsArrayAsync<TItem, TResult>(this IEnumerable<TItem> source, Func<TItem, ValueTask<TResult>> selector)
+        {
+            var builder = ArrayBuilder<TResult>.GetInstance();
```

Comment:
```suggestion
            using var _ = ArrayBuilder<TResult>.GetInstance(out var builder);
```

this ensures we cleanup with cancelaltion.

Reply:
remember to use ToImmutable below if you do this.

Reply:
Fixed.

Reply:
Actually not fixed: this is in the compiler layer and linked into workspaces, but it seems those helpers aren't in the compiler layer. I don't understand any of this to know why.

Reply:
gotcha.  wacky.